sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py

```diff
@@ -537,7 +543,7 @@ def __len__(self):
     def _from_list(self, messages, parent_span=None):
         # type: (Iterable[ServiceBusMessage], AbstractSpan) -> None
```

Comment:
```suggestion
        # type: (Iterable[Union[ServiceBusMessage, dict]], AbstractSpan) -> None
```

Reply:
So I actually asked Anna about this and she mentioned that we don't want to explicitly mention that we accept dicts, b/c we want to discourage users from passing in dicts and not following the "regular flow". Here's a previous comment from a different commit that addresses this: https://github.com/Azure/azure-sdk-for-python/pull/14807#discussion_r516872709

Reply:
thanks for the breadcrumb, fair enough to me!

Reply:
hmm I wouldn't say we want to _discourage_ users from passing dicts...
And actually now that I think on it, we don't want to give users static-analysis errors when they attempt to use dicts.
It's more that they're not currently widely documented in our API type hints where we do support it....
It is easy enough to add later - additionally, I think it would be better to support a generic `typing.Mapping` type, rather than an explicit `dict` (this would need to be updated in the `isinstance` check as well).

Reply:
If we chose not to add it to this PR, we should open an issue to address it later.

Reply:
sounds good, I updated the `isinstance` to check for mapping rather than dict. I'll open an issue for this.

Reply:
note: discussed with Anna, and will be removing check with isinstance using Mapping. other validation will need to be added to check that the dict sent in conforms to protocol. also will be added to above issue